core/dbt/adapters/factory.py

```diff
@@ -89,7 +89,8 @@ def load_plugin(self, name: str) -> Type[Credentials]:
     def register_adapter(self, config: AdapterRequiredConfig) -> None:
         adapter_name = config.credentials.type
         adapter_type = self.get_adapter_class_by_name(adapter_name)
-
+        adapter_version = import_module(f".{adapter_name}.__version__", "dbt.adapters").version
+        fire_event(AdapterRegistered(adapter_name=adapter_name, adapter_version=adapter_version))
```

Comment:
Took it for a spin, and it looks like there are some formatting inconsistencies between the adapter and dbt versions:

```sh
04:35:56  Running with dbt=1.6.0-b4
04:35:56  target not specified in profile 'postgres', using 'default'
04:35:57  Registered adapter: postgres, version: 1.6.0b4 # no - between 1.6.0 and b4
```

We use [dbt.semver.VersionSpecifier.from_version_string](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/version.py#L38) to [format the core version](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/version.py#L236), for example:  
```python
import dbt.semver
adapter_version_specifier = dbt.semver.VersionSpecifier.from_version_string(adapter_version)
fire_event(AdapterRegistered(adapter_name=adapter_name, adapter_version=str(adapter_version_specifier)))
```